array_api_strict/_array_object.py

```diff
@@ -234,6 +234,8 @@ def _check_device(self, other):
         elif isinstance(other, Array):
             if self.device != other.device:
                 raise ValueError(f"Arrays from two different devices ({self.device} and {other.device}) can not be combined.")
+        else:
+            raise TypeError(f"Cannot combine an Array with {type(other)}.")
```

Comment:
When does it happen that we get here with `other` being neither a scalar nor an array? My expectation was somehow that  this wouldn't happen

Maybe worth adding a comment given that it isn't immediately obvious to me now and probably also not obvious to readers from the future.

Reply:
Good catch, thank you! This is interference from https://github.com/data-apis/array-api-strict/pull/103. 
Should probably decide on that PR one way or another and rebase this one.